### PR TITLE
Add --additional-features option

### PR DESCRIPTION
**Pull request by stuartleeks**
Initial work to enable additional features to be specified via the CLI

### CODE REVIEW

File: src/spec-node/devContainersSpecCLI.ts [hunk 1]
```diff
@@ -735,8 +746,9 @@ async function readConfiguration({
 			};
 		}
 
-		const needsFeaturesConfig = includeFeaturesConfig || (includeMergedConfig && (!container || !experimentalImageMetadata));
-		const featuresConfiguration = needsFeaturesConfig ? await readFeaturesConfig(params, pkg, configuration.config, extensionPath, skipFeatureAutoMapping) : undefined;
+		const additionalFeatures = additionalFeaturesJson ? jsonc.parse(additionalFeaturesJson) as Record<string, string | boolean | Record<string, string | boolean>> : undefined;
+		const needsFeaturesConfig = includeFeaturesConfig || additionalFeatures || (includeMergedConfig && (!container || !experimentalImageMetadata));
```
Comment (chrmarti) [inline src/spec-node/devContainersSpecCLI.ts:750]: The features config shouldn't be needed when the container already exists. The container will have the config from the additional features on its devcontainers.metadata label. (Fetching the features config can be expensive.) Please remove the `additionalFeatures` from the condition.

File: src/spec-configuration/containerFeaturesConfiguration.ts [hunk 2]
```diff
@@ -491,19 +491,38 @@ async function prepareOCICache(dstFolder: string) {
 	return ociCacheDir;
 }
 
-function featuresToArray(config: DevContainerConfig): DevContainerFeature[] | undefined {
-	if (!config.features) {
+function featuresToArray(config: DevContainerConfig, additionalFeatures?: Record<string, string | boolean | Record<string, string | boolean>>): DevContainerFeature[] | undefined {
```
Comment (chrmarti) [inline src/spec-configuration/containerFeaturesConfiguration.ts:494]: Please make `additionalFeatures` a mandatory argument here and everywhere else where we want to make sure we don't forget pass it on.

**Review by chrmarti — CHANGES_REQUESTED** [review]
Looks great. Left two comments for change / discussion. Thanks!

File: src/spec-node/devContainersSpecCLI.ts (hunk 1 above)
Comment (stuartleeks) [reply src/spec-node/devContainersSpecCLI.ts:756]: Good point - updated

File: src/spec-configuration/containerFeaturesConfiguration.ts (hunk 2 above)
Comment (stuartleeks) [reply src/spec-configuration/containerFeaturesConfiguration.ts]: Done 😄

**Comment by chrmarti** [comment]
> @chrmarti - I've pushed an update that hopefully addresses your feedback

@stuartleeks Still seeing the additional clause in `readConfiguration`. I think we do without it to avoid reading the entire features config when the container already exists:
image

**Comment by stuartleeks** [comment]
Oops! I'm not sure what happened there - sorry!

I have reapplied the change to remove `additionalFeatures` from the `needsFeaturesConfig` expression.

**Pull request merged**